README.md

```diff
@@ -1,4 +1,137 @@
 
-This is an example of local matching service that could be used by relaying parties as a reference when building their matching service.
+This is an example local matching service built by the GOV.UK Verify team to demonstrate how a local matching service works and help services building their own. Every government service connecting to Verify must build their own local matching service.
 
+:warning: _This is an example implementation and is not production-ready._ :warning:
+
+For more guidance on the role of a local matching service and how to build one, refer to the:
+* [Verify technical guide](http://alphagov.github.io/rp-onboarding-tech-docs/pages/ms/ms.html)
+* [development stories for this example](https://github.com/alphagov/verify-local-matching-service-example/projects/1)
+
+## Background
+
+This example uses a simple database and matching strategy. It has not been designed to be used in front of a real database.
+
+### Database
+
+This example local matching service uses a SQL database. Refer to the [database schema](/docs/schema.png) for details.
+
+All SQL has been written for Postgres and has not been tested with any engine.
+
+### Matching strategy
+
+This example implementation follows a simplified matching strategy from DWP.
+
+The example uses the following attributes:
+
+* firstname
+* surname
+* date of birth
+* postcode
+* National Insurance number
+
+**Matching cycles**
+
+This example follows [Cycle 0, Cycle 1 and Cycle 3 match cycles](http://alphagov.github.io/rp-onboarding-tech-docs/pages/ms/msWorks.html).
+
+It will first run Cycle 0. The Matching Service Adapter will provide a hashed persistent identifier (PID). The cycle will then check the PID can be matched to an existing PID in the database. This cycle only works if the user has previously verified their identity with the same identity provider.
+
+If Cycle 0 fails, it will run Cycle 1 using surname and date of birth. The example will disambiguate similar matches with postcode and then firstname. All four attributes need to match in order to return `match`.
+
+If the request contains a National Insurance number, the example will run Cycle 3. Using National Insurance number as a Cycle 3 attribute was an arbitrary decision for the purposes of this example. When building your own local matching service, you should use other attributes.
+
+If the example finds a match in Cycles 1 or 3 it will store the PID in the database.
+
+**Caveats**
+
+This example local matching service will only return `match` or `no match`. It does not provide any fuzzy matching, for example matching `ONeill` to `O'Neill` or `Will` to `William`. The example does not provide any means of logging or investigating results.
+
+The example ignores any attributes where `verified` is set to `false`. This would be a suitable starting point for any service with an existing dataset expected to include most new users.
+
+The example will ignore any verified historical
```

Comment:
It only ignores any unverified attribute as mentioned above. We are not ignoring the "verified" historical attributes.